File: client/src/models/Post.ts
import { Profile } from "./Profile";

export interface IPost {
  id: string;
  title: string;
  date: Date;
  description: string;
  category: string;
  image: string | null;
  creator?: Profile;
  likes: Profile[];
  isLiked: boolean;
}

export class Post implements IPost {
  constructor(init: PostFormValues) {
    this.id = init.id!;
    this.title = init.title;
    if (init.description) this.description = init.description;
    this.category = init.category;
  }

  id: string;
  title: string;
  date: Date = new Date();
  description: string = "";
  category: string;
  image: string | null = null;
  creator?: Profile;
  likes: Profile[] = [];
  isLiked: boolean = false;
}

export class PostFormValues {
  id?: string = undefined;
  title: string = "";
  description?: string = "";
  photo?: Blob;
  category: string = "";
}

export enum Category {
  Humor = "Humor",
  Food = "Food",
  News = "News",
  Animals = "Animals",
  Sports = "Sports",
}

File: Domain/Post.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Post
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public string Image { get; set; }
        public AppUser Creator { get; set; }
        public ICollection<Like> Likes { get; set; } = new List<Like>();
        public ICollection<Photo> Photos { get; set; }
        // likes + comments
    }
}